Lib/test/test_netrc.py

```diff
@@ -109,62 +109,68 @@ def test_comment_at_end_of_machine_line_pass_has_hash(self):
     def test_security(self):
         # This test is incomplete since we are normally not run as root and
         # therefore can't test the file ownership being wrong.
-        d = os_helper.TESTFN
-        os.mkdir(d)
-        self.addCleanup(os_helper.rmtree, d)
-        fn = os.path.join(d, '.netrc')
-        with open(fn, 'wt') as f:
-            f.write("""\
-                machine foo.domain.com login bar password pass
-                default login foo password pass
-                """)
-        with os_helper.EnvironmentVarGuard() as environ:
-            environ.set('HOME', d)
-            os.chmod(fn, 0o600)
-            nrc = netrc.netrc()
-            self.assertEqual(nrc.hosts['foo.domain.com'],
-                             ('bar', None, 'pass'))
-            os.chmod(fn, 0o622)
-            self.assertRaises(netrc.NetrcParseError, netrc.netrc)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with os_helper.change_cwd(tmpdir):
+                d = os_helper.TESTFN
+                os.mkdir(d)
+                self.addCleanup(os_helper.rmtree, d)
+                fn = os.path.join(d, '.netrc')
```

Comment:
You create a temporary directory. Why do you still create a second TESTFN temporary directory? Just use tmpdir, no?

```suggestion
                fn = os.path.join(tmpdir, '.netrc')
```

Same remark for the other functions.

Reply:
Thanks. Fixed.